lib/AST/StmtPrinter.cpp

```diff
@@ -1903,6 +1903,13 @@ void StmtPrinter::VisitInteropTypeBoundsAnnotation(
   }
 }
 
+void StmtPrinter::VisitBoundsCastExpr(BoundsCastExpr *Node) {
```

Comment:
I don't think this will print the correct output.    It won't be what was originally written.